src/main/sensors/gyro.c

```diff
@@ -544,12 +549,33 @@ bool gyroInit(void)
             gyroConfigMutable()->gyro_to_use = GYRO_CONFIG_USE_GYRO_1;
             detectedSensors[SENSOR_INDEX_GYRO] = gyroSensor1.gyroDev.gyroHardware;
             sensorsSet(SENSOR_GYRO);
+        }
+    }
 
+    if (gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH && detectionFlags != (DETECTED_GYRO_1 | DETECTED_GYRO_2)) {
```

Comment:
This should probably go above the check for the gyros being of a different type, and the check should not be done if only one gyro has been detected.

Reply:
Yeah I did have a quick look at that, I think you're right.  Let me fix that now for you.